framework/test/unit/modules/interoperability/mainchain/commands/message_recovery.spec.ts

```diff
@@ -647,14 +627,256 @@ describe('Mainchain MessageRecoveryCommand', () => {
+
+			await expect(command['_applyRecovery'](context)).resolves.toBeUndefined();
+
+			expect(context.eventQueue.restoreSnapshot).toHaveBeenCalledWith(99);
+			expect(context.stateStore.restoreSnapshot).toHaveBeenCalledWith(10);
+		});
+
+		it('should log event and restore the state/event before calling execute when execute fails', async () => {
+			(((command['ccCommands'].get(defaultCCM.module) as BaseCCCommand[]).find(
+				com => com.name === defaultCCM.crossChainCommand,
+			) as BaseCCCommand).execute as jest.Mock).mockRejectedValue('error');
+			let eventQueueCount = 0;
+			let stateStoreCount = 0;
+			jest.spyOn(context.eventQueue, 'createSnapshot').mockImplementation(() => {
+				eventQueueCount += 1;
+				return eventQueueCount;
+			});
+			jest.spyOn(context.stateStore, 'createSnapshot').mockImplementation(() => {
+				stateStoreCount += 1;
+				return stateStoreCount;
+			});
+			jest.spyOn(context.eventQueue, 'restoreSnapshot');
+			jest.spyOn(context.stateStore, 'restoreSnapshot');
+
+			await expect(command['_applyRecovery'](context)).resolves.toBeUndefined();
+
+			expect(context.eventQueue.restoreSnapshot).toHaveBeenCalledWith(2);
+			expect(context.stateStore.restoreSnapshot).toHaveBeenCalledWith(2);
+			expect(
+				(command['interoperableCCMethods'].get('token') as BaseCCMethod)
+					.afterCrossChainCommandExecute as jest.Mock,
+			).toHaveBeenCalledTimes(1);
+		});
+
+		it('should log event when command afterCrossChainCommandExecute fails', async () => {
+			((command['interoperableCCMethods'].get('token') as BaseCCMethod)
+				.afterCrossChainCommandExecute as jest.Mock).mockRejectedValue('error');
+
+			await expect(command['_applyRecovery'](context)).resolves.toBeUndefined();
+
+			expect(context.eventQueue.getEvents()).toHaveLength(1);
+			expect(command['events'].get(CcmProcessedEvent).log).toHaveBeenCalledWith(
+				expect.anything(),
+				context.ccm.receivingChainID,
+				context.ccm.sendingChainID,
+				{
+					ccmID: expect.any(Buffer),
+					code: CCMProcessedCode.INVALID_CCM_AFTER_CCC_EXECUTION_EXCEPTION,
+					result: CCMProcessedResult.DISCARDED,
+				},
+			);
+		});
+
+		it('should restore the original state/event when command afterCrossChainCommandExecute fails', async () => {
+			((command['interoperableCCMethods'].get('token') as BaseCCMethod)
+				.afterCrossChainCommandExecute as jest.Mock).mockRejectedValue('error');
+			jest.spyOn(context.eventQueue, 'createSnapshot').mockReturnValue(99);
+			jest.spyOn(context.stateStore, 'createSnapshot').mockReturnValue(10);
+			jest.spyOn(context.eventQueue, 'restoreSnapshot');
+			jest.spyOn(context.stateStore, 'restoreSnapshot');
+
+			await expect(command['_applyRecovery'](context)).resolves.toBeUndefined();
+
+			expect(context.eventQueue.restoreSnapshot).toHaveBeenCalledWith(99);
+			expect(context.stateStore.restoreSnapshot).toHaveBeenCalledWith(10);
+		});
+
+		it('call all the hooks if defined', async () => {
```

Comment:
description is a bit vague,  maybe something like `should log success event when all the hooks pass` ?